src/client/game.cpp

```diff
@@ -1757,6 +1759,9 @@ void Game::updateDebugState()
 		hud->disableBlockBounds();
 	if (!has_debug)
 		draw_control->show_wireframe = false;
+
+	// noclip
+	draw_control->allow_noclip = m_cache_enable_noclip && client->checkPrivilege("noclip");
```

Comment:
Why would this depend on the cached setting value? This may chance at any time, hence I wonder why this was introduced.